### PR TITLE
Transferred writing_functions.adoc

### PR DESCRIPTION
**Pull request by daniakhanpurdue**
Moved writing_functions.adoc from the old examples book to the new one in the Python section

### CODE REVIEW

File: content/modules/python/pages/writing_functions.adoc [hunk 1]
```diff
@@ -0,0 +1,462 @@
...
+int: The number of words in the sentence
+----
+
+In addition, if you are coding in a tool like VSCode, for example, you may gain the ability to hover over a function and see its docstring and other information.
+
+It is good practice to write docstrings for every function you write.
+
+== annotations
+
+Another "thing" you may have noticed from our `word_count` function if you've ever used Python in the past. In the _signature_ of our function `def` `word_count(sentence: str) -> int`, we have some extra, _not_ required information in the form of _function annotations_. Specifically, you could write the `word_count` function like this, but it would function just the same:
```
Comment (lacykq) [inline content/modules/python/pages/writing_functions.adoc:381]: In the `word_count` example, you may notice that the format is a little bit different from the rest of the examples used on this page. The `word_count` function consists extra (not necessary) information. This is a form of type annotations. 

You can shorten the `word_count` function, and it still will do the same:

File: content/modules/python/pages/writing_functions.adoc [hunk 2]
```diff
@@ -0,0 +1,462 @@
...
+        helper function to strip punctuation.
+        """
+        return sentence.translate(str.maketrans('', '', string.punctuation))
+    
+    sentence_no_punc = _strip_punctuation(sentence)
+    result = len(sentence_no_punc.split())
+    return result
+----
+
+Here, we do not specify that `sentence` should be a `str` or that the returned result should be an `int`. When we _do_ annotate functions, it is purely a way to add metadata to our function. In large projects, function annotations are recommended. Although Python does not strictly enforce type annotations, packages like `mypy` can be added to a deployment scheme to strictly enforce it.
```
Comment (lacykq) [inline content/modules/python/pages/writing_functions.adoc:408]: In the example above, the expected types for `sentence` and output are removed. Annotating a function is a way to add data about the function (metadata) to the function. Type annotations are strongly encouraged in large or collaborated projects. 

NOTE: Packages like `mypy` will enforce type annotations. They can be tools to help you develop good code practices.

File: content/modules/python/pages/writing_functions.adoc [hunk 3]
```diff
@@ -0,0 +1,462 @@
...
+
+== decorators
+
+=== Examples
+
+*Write a function called get_filename_from_url that, given a url to a file, like https://image.shutterstock.com/image-vector/cute-dogs-line-art-border-260nw-1079902403.jpg returns the filename with the extension.*
```
Comment (lacykq) [inline content/modules/python/pages/writing_functions.adoc:414]: No need for this whole sentence to be bold. 

The function below is called `get_filename_from_url` that accepts a URL to a file as an argument and returns the file name with the extension. 

Use this [URL](https://image.shutterstock.com/image-vector/cute-dogs-line-art-border-260nw-1079902403.jpg) to a picture of dogs for our example.

File: content/modules/python/pages/writing_functions.adoc [hunk 4]
```diff
@@ -0,0 +1,462 @@
...
+    Args:
+        url (str): The url of the file.
+
+    Returns:
+        str: A string with the filename, including the file extension.
+    """
+    return os.path.basename(urlparse(url).path)
+----
+
+*Write a function that, given a url to an image, and a full path to a directory, saves the image to the provided directory. By default, have the function save the images to the user's home directory in a unix-like operating system.*
```
Comment (lacykq) [inline content/modules/python/pages/writing_functions.adoc:434]: The function below is called `scrape_image`. It accepts a URL to an image and a full path to a directory as arguments. The function saves the image to the provided directory. If a path is not provided, (by default) the function will save the image to the user's home directory (in a UNIX-like operating system).

**Comment by lacykq** [comment]
@daniakhanpurdue @lsukova  

Just wanted to clarify, I am not sure if you both are working from the same document in an ordered way, but both of you have a pull request for the same page. 

@naomim01 made some comments on Drew's pull request, and I just made some comments on Dania's above. 

What would the best way to communicate to make sure that aware of which pages you are making pull requests on so that we are are all not duplicating the same work


Tagged @kevinamstutz and @dglass19 to keep them in the loop

**Comment by lsukova** [comment]
@lacykq I've been reassigning each goal in the "Issues" page to myself when I start working on it. I believe @kevinamstutz demonstrated this idea during one of the introductory meetings.

**Comment by daniakhanpurdue** [comment]
That’s my bad, it must have gone over my head when he explained it – I’ve just been checking the pull requests to see if it was already done or not -  I can assign those issues to myself from now on, my bad!

Sorry for any inconvenience,
Dania Khan
Purdue University – May 2022
Manufacturing Engineering Technology

**Pull request closed without merging**